infrastructure/smart-contracts/test/stakingv2/SubmittingAndClaiming.mjs

```diff
@@ -47,17 +47,28 @@ export function SubmittingAndClaiming(deployInfrastructure, poolConfigurations)
 			const stakingPoolAddress = await poolFactory.connect(addr2).getPoolAddress(0);
 			console.log("stakingPoolAddress", stakingPoolAddress);
 
-			// Mint 100 keys to addr2 & stake
-			// Changed to 100 as 2 keys would not generate a reward thus causing a divide by zero error for number of eligible claimers
-			const keysToMintForAddr2 = 100n;
-			const addr2KeyMintPrice = await nodeLicense.price(keysToMintForAddr2, "");
-			await nodeLicense.connect(addr2).mint(keysToMintForAddr2, "", {value: addr2KeyMintPrice});
-			const addr2MintedKeyIds = [];
+			// Mint in batches of 100 keys to addr2 & stake
+			// Changed to 10_000 to have a high enough chance to get a winning key
+			const keysToMintForAddr2 = 10_000n;
+			const mintsPerLoop = 100n;
+			for (let i = keysToMintForAddr2; i > 0; i -= mintsPerLoop) {
+				let mintQuantity = mintsPerLoop;
+				if (i < mintsPerLoop) {
+					mintQuantity = i;
+				}
+				const addr2KeyMintPrice = await nodeLicense.price(mintQuantity, "");
+				await nodeLicense.connect(addr2).mint(mintQuantity, "", {value: addr2KeyMintPrice});
+			}
```

Comment:
Just FYI, there is a helper function in the test utils where you can do this in a single line:

`const addr2MintedKeyIds = await mintBatchedLicenses(keysToMintForAddr2, nodeLicense, addr2);`